CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+You can run this check locally:
+
+```
+cd terraform-docs-samples
+make docker_test_lint
+```
+
+## Integration Testing
+
+Integration tests are used to verify that the samples are actuatable
+by Terraform. Tests are dynamically discovered based on directories in
+repo root and executed using the [blueprint-test](https://pkg.go.dev/github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test) framework.
+
+### Test Environment
+
+The easiest way to test the samples are in isolated test projects. The
+setup for projects are defined in [test/setup](./test/setup/)
+directory.
+
+To use this setup, you need to authenticate gcloud with an identity that has
+Project Creator access on a folder and the Billing Account User role on a billing account.
+
+Set the following environment variables.
+```
+export TF_VAR_org_id="your_org_id"
+export TF_VAR_folder_id="your_folder_id"
+export TF_VAR_billing_account="your_billing_account_id"
+```
+
+Ensure that you are locally authenticated using gcloud:
+
+```
+gcloud auth application-default login
+```
+
+With these settings in place, you can prepare a test project using Docker:
+
+```
+cd terraform-docs-samples
+make docker_test_prepare
+```
+
+### Noninteractive Execution
+
+Run `make -s docker_test_sample SAMPLE=${SAMPLE_NAME}` to test a sample
+noninteractively. This will initialize, apply, verify and destroy the
+specified sample.
+
+Example:
+```
+make -s docker_test_sample SAMPLE=storage_new_bucket
+```
+
+Sample output:
+
+```
+ make -s docker_test_sample SAMPLE=storage_new_bucket
+go: downloading github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test v0.3.0
+...
+=== RUN   TestSamples
+TestSamples terraform.go:216: Loading env vars from setup ../setup
+...
+2022/10/28 02:30:16 Running stage init
+TestSamples/1/storage_new_bucket retry.go:91: terraform [init -upgrade=false]
+...
+2022/10/28 02:30:17 Running stage apply
+TestSamples/1/storage_new_bucket retry.go:91: terraform [apply -input=false -auto-approve -lock=false]
+...
+2022/10/28 02:30:21 Running stage verify
+TestSamples/1/storage_new_bucket cmd.go:103: Running terraform with args [plan -input=false -detailed-exitcode -lock=false]
+...
+2022/10/28 02:30:23 Running stage teardown
+TestSamples/1/storage_new_bucket retry.go:91: terraform [destroy -auto-approve -input=false -lock=false]
+...
+TestSamples/1/storage_new_bucket command.go:185: Destroy complete! Resources: 3 destroyed.
+TestSamples/1/storage_new_bucket command.go:185:
+--- PASS: TestSamples (17.12s)
+    --- PASS: TestSamples/1/storage_new_bucket (13.95s)
+PASS
+ok      github.com/terraform-google-modules/terraform-docs-samples/test/integration     17.203s
+```
+
+### Interactive Execution
+
+Interactive execution is useful if you want to iteratively test a sample
+without destroying resources automatically.
+
+1. Run `make docker_run` to start the testing Docker container in
+   interactive mode.
+
```

Comment:
Here can we add:

The command prompt changes to `[root@8d7469507cf8 workspace]#`.

Reply:
done